### PR TITLE
Zero config database support

### PR DESCRIPTION
**Pull request by andriisoldatenko**
fix astronomer/issues#707

### CODE REVIEW

**Comment by andriisoldatenko** [comment]
practical test:
```
./bin/reset-local-dev
$ astronomer git:(zero-config-database-support) ✗ helm list
NAME            REVISION        UPDATED                         STATUS          CHART                           APP VERSION     NAMESPACE
astronomer      1               Sat Jan 25 00:12:32 2020        DEPLOYED        astronomer-platform-0.11.0      0.11.0          astronomer
```

```
(base) ➜  astronomer git:(zero-config-database-support) ✗ k get pods -n astronomer
NAME                                                READY   STATUS    RESTARTS   AGE
astronomer-cli-install-6546ccf4c8-bt7n4             1/1     Running   0          6m15s
astronomer-commander-59fdbf9ff6-ccmpl               1/1     Running   0          6m15s
astronomer-houston-744f59c9c6-msw5l                 1/1     Running   0          6m15s
astronomer-kubed-85f87748c6-x2kb8                   1/1     Running   0          6m15s
astronomer-nginx-7dc54d9c7d-2kpwx                   1/1     Running   0          6m15s
astronomer-nginx-default-backend-7956f565dd-cx7p6   1/1     Running   0          6m15s
astronomer-orbit-59c7d57c6c-n2pjq                   1/1     Running   0          6m15s
astronomer-postgresql-0                             1/1     Running   0          6m15s
astronomer-prisma-66cdd4b5c5-llbx9                  1/1     Running   0          6m15s
astronomer-registry-0                               1/1     Running   0          6m15s
```

**Comment by andriisoldatenko** [comment]
@schnie it's doesn't work, because in my original PR i did some change in ariflow-chart...
Right now we are using separate repo...not sure how to combine it better all together.
```diff
diff --git templates/secrets/metadata-connection-secret.yaml templates/secrets/metadata-connection-secret.yaml
index 4acf0c3..491f0a2 100644
--- templates/secrets/metadata-connection-secret.yaml
+++ templates/secrets/metadata-connection-secret.yaml
@@ -2,7 +2,7 @@
 ## Airflow Metadata Secret
 #################################
 {{- if (and .Values.data.metadataConnection (not .Values.data.metadataSecretName)) }}
-{{- $host := .Values.data.metadataConnection.host | default (printf "%s-%s.%s.svc.cluster.local" .Release.Name "postgresql" .Release.Namespace) }}
+{{- $host := .Values.data.metadataConnection.host | default (printf "%s-%s" .Release.Name "postgresql") }}
 kind: Secret
 apiVersion: v1
 metadata:
```

**Comment by andriisoldatenko** [comment]
```
2020-01-25 00:15:57.511 1 WARNING lookup failed: astronomer-postgresql: result=0
2020-01-25 00:15:57.511 1 LOG S-0x563a156349b0: quasaric-wormhole-0751-metadata/quasaric_wormhole_0751_airflow@(bad-af):0 closing because: server dns lookup failed (age=0)
2020-01-25 00:15:59.955 1 LOG Stats: 0 xacts/s, 0 queries/s, in 0 B/s, out 0 B/s, xact 0 us, query 0 us wait time 0 us
2020-01-25 00:16:00.603 1 LOG C-0x563a1562fe90: pgbouncer/quasaric_wormhole_0751_airflow@127.0.0.1:59214 login attempt: db=pgbouncer user=quasaric_wormhole_0751_airflow tls=no
2020-01-25 00:16:00.603 1 LOG C-0x563a1562fe90: pgbouncer/quasaric_wormhole_0751_airflow@127.0.0.1:59214 closing because: client close request (age=0)
2020-01-25 00:16:01.206 1 LOG C-0x563a1562fe90: (nodb)/(nouser)@10.244.0.1:44188 closing because: client unexpected eof (age=0)
^[[A^[[A2020-01-25 00:16:06.703 1 LOG C-0x563a1562fe90: pgbouncer/quasaric_wormhole_0751_airflow@127.0.0.1:59272 login attempt: db=pgbouncer user=quasaric_wormhole_0751_airflow tls=no
2020-01-25 00:16:06.704 1 LOG C-0x563a1562fe90: pgbouncer/quasaric_wormhole_0751_airflow@127.0.0.1:59272 closing because: client close request (age=0)
2020-01-25 00:16:06.968 1 LOG C-0x563a1562fe90: (nodb)/(nouser)@10.244.0.1:44234 closing because: client unexpected eof (age=0)
2020-01-25 00:16:09.104 1 LOG C-0x563a1562fe90: (nodb)/(nouser)@10.244.0.30:41929 closing because: client unexpected eof (age=0)
2020-01-25 00:16:10.603 1 LOG C-0x563a1562fe90: pgbouncer/quasaric_wormhole_0751_airflow@127.0.0.1:59320 login attempt: db=pgbouncer user=quasaric_wormhole_0751_airflow tls=no
2020-01-25 00:16:10.603 1 LOG C-0x563a1562fe90: pgbouncer/quasaric_wormhole_0751_airflow@127.0.0.1:59320 closing because: client close request (age=0)
2020-01-25 00:16:11.206 1 LOG C-0x563a1562fe90: (nodb)/(nouser)@10.244.0.1:44294 closing because: client unexpected eof (age=0)
2020-01-25 00:16:12.571 1 WARNING lookup failed: astronomer-postgresql: result=0
2020-01-25 00:16:12.571 1 LOG S-0x563a156349b0: quasaric-wormhole-0751-metadata/quasaric_wormhole_0751_airflow@(bad-af):0 closing because: server dns lookup failed (age=0)
```

**Comment by andriisoldatenko** [comment]
@schnie @sjmiller609 what i'm doing wrong here?
looks like [pg-bouncer](https://github.com/astronomer/airflow-chart/tree/master/templates/pgbouncer) from doesn't have access to this potgresql pod... 🤔 
any ideas?

**Comment by schnie** [comment]
Ah, probably because the bootstrap secret needs to have `.<namespace>.svc.cluster.local` instead of just the service name since it's possible the airflows go into a different namespace.

**Comment by andriisoldatenko** [comment]
@schnie fixed 🎉 
```
$ kubectl get pods -n astronomer
NAME                                                READY   STATUS    RESTARTS   AGE
astronomer-cli-install-6546ccf4c8-dd8n8             1/1     Running   0          12m
astronomer-commander-59fdbf9ff6-8xshs               1/1     Running   0          12m
astronomer-houston-744f59c9c6-gtbs5                 1/1     Running   0          12m
astronomer-kubed-7bf69d46d-fkh5r                    1/1     Running   0          12m
astronomer-nginx-7dc54d9c7d-psnlm                   1/1     Running   0          12m
astronomer-nginx-default-backend-7956f565dd-twl22   1/1     Running   0          12m
astronomer-orbit-59c7d57c6c-q7rz9                   1/1     Running   0          12m
astronomer-postgresql-0                             1/1     Running   0          12m
astronomer-prisma-66cdd4b5c5-d5qxl                  1/1     Running   0          12m
astronomer-registry-0                               1/1     Running   0          12m

$ kubectl get pods -n astronomer-elementary-flyby-3027
NAME                                               READY   STATUS      RESTARTS   AGE
elementary-flyby-3027-cleanup-1580139000-pfmxb     0/1     Completed   0          2m47s
elementary-flyby-3027-flower-855d6c57fc-pfx8d      1/1     Running     0          3m6s
elementary-flyby-3027-pgbouncer-568cd69c76-dcq7h   2/2     Running     0          3m6s
elementary-flyby-3027-redis-0                      1/1     Running     0          3m6s
elementary-flyby-3027-scheduler-57b9bf6cb5-7l2w7   2/2     Running     0          3m6s
elementary-flyby-3027-statsd-d64fff94d-xb4m2       1/1     Running     0          3m6s
elementary-flyby-3027-webserver-c8854b7f-5slcn     1/1     Running     0          3m6s
elementary-flyby-3027-worker-7889f97cd6-tcgcb      1/1     Running     0          3m6s

```

File: values.yaml [hunk 1]
```diff
@@ -301,3 +301,11 @@ kubed:
...
       cpu: "2"
       memory: "1024Mi"
+
+#################################
+## Postgresql Config
+#################################
+# Not recommended to use an in-cluster
+# database for production.
+postgresql:
+  enabled: false
```
Comment (andriisoldatenko) [inline values.yaml]: should we enable by default ?
cc @schnie

**Pull request merged**